ui/packages/catalog-ui-search/src/main/webapp/js/widgets/cesium.base.line.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+
+const Marionette = require('marionette');
+const Cesium = require('cesium');
+const _ = require('underscore');
+const wreqr = require('wreqr');
+const NotificationView = require('./notification.view');
+const DrawingController = require('./drawing.controller');
+
+"use strict";
+
+const CAMERA_MAGNITUDE_THRESHOLD = 8000000;
+
+const capitalize = (value) => value.charAt(0).toUpperCase() + value.slice(1);
+
+const getCurrentMagnitudeFromView = (view) => {
+    return view.map.camera.getMagnitude();
+}
+
+const needsRedraw = (view) => {
+    const currentMagnitude = getCurrentMagnitudeFromView(view);
+
+    if (view.cameraMagnitude < CAMERA_MAGNITUDE_THRESHOLD && currentMagnitude > CAMERA_MAGNITUDE_THRESHOLD) {
+        return true;
+    }
+    if (view.cameraMagnitude > CAMERA_MAGNITUDE_THRESHOLD && currentMagnitude < CAMERA_MAGNITUDE_THRESHOLD) {
+        return true;
+    }
+
+    return false;
+}
+
+class GeometryRenderView extends Marionette.View {
+  constructor(options) {
+    super(...options);
+
+    Object.assign(this, {...options});
```

Comment:
❓ If options is an object passed in, do we need to create a new object from it?

Reply:
I'm not creating a new object from it, just assigning the values to `this` since the super class doesn't assign the fields I need (like `map`, `drawHelper`, etc.).